native/common/jp_gc.cpp

```diff
@@ -237,10 +245,24 @@ void JPGarbageCollection::onEnd()
 		Py_ssize_t pred = current + 2 * (current - last);
 		last = current;
 		if ((Py_ssize_t) pred > (Py_ssize_t) limit)
+		{
 			run_gc = 2;
+            limit = high_water + (high_water>>3) + 8 * (current - last);
+        }
 
-		//		printf("consider gc %d (%ld, %ld, %ld, %ld) %ld\n", run_gc,
-		//				current, low_water, high_water, limit, limit - pred);
+#if 0
```

Comment:
Did you forget to commit something? This is unconditional true, I think it should be activated upon instrumentation, right?

It'd also be nice to use a consistent formatting (tabs vs spaces)

Reply:
The instrumentation is to barf out all of the python gc calls and all the java calls.   I am not sure anyone is likely to want that unless they are trying to debug the process.   So I currently have it conditionally compiled as false all the time.

I will try to fix up the tabs.   The problem is that some files we have are currently tab spaced and others are spaced only and I have to keep switching the settings on vim based on which file I am editing.

Reply:
I've found it rather painful as well while working on extension support. It got to the point where I've said to hell with it and reformatted entire files and decided to just deal with fixing it later.

Reply:
We should come up with some codestyle probably. Did you choose spaces over tabs or vice versa @astrelsky?

Reply:
> We should come up with some codestyle probably. Did you choose spaces over tabs or vice versa @astrelsky?

My ide settings were set to use this formatter so it only took me two clicks to reformat a file. https://github.com/NationalSecurityAgency/ghidra/blob/master/eclipse/GhidraEclipseFormatter.xml

I learned java from working with Ghidra so I lean towards their code style with minor divergence. In many ways it directly conflicts with the styles used in jpype. I've become very nitpicky about formatting.

To actually answer your question, I used tabs (width of 4 spaces). At the end of the day I'll follow whatever code style you set up. I was aware I was creating extra work for myself by applying the formatter when I did it.

I don't think I should have any say here. As long as it's consistent I'll follow it :joy:

Reply:
It was so much easier when netbeans had native C++ support.   My only requirement is please no K&R type brace styles.   My preferences is always the GNU coding style.  https://www.gnu.org/prep/standards/html_node/Formatting.html

Reply:
> It was so much easier when netbeans had native C++ support.
> 
> 

If you're wanting something that supports all 3 languages I've used vscode with good results. Setup can be a hassle, especially with java, but I have managed to have a python debugger, java debugger and c++ debugger all attached at the same time in the same vscode instance while running the jpype python tests through vscode. I'm not necessarily recommending it, because project configuration for java and c++ can be extremely finicky, just putting it out there.

Reply:
Before the transition to Apache I had Netbeans configured to all of the languages, with debugging, autoformat, etc.    It was very easy to compile and test.   After Apache C++ extensions decayed to the point they couldn't even open the files and reformat them.   Thus I have been forced back to stone knives and bear skins (well using vim and doing everything from command line).   It has been a serious reduction in what I can do as refactoring is out of the question with such a setup.  

It appears that netbeans has finally readded C++ support, so I will try to get it spun up again.

Reply:
> Before the transition to Apache I had Netbeans configured to all of the languages, with debugging, autoformat, etc.    It was very easy to compile and test.   After Apache C++ extensions decayed to the point they couldn't even open the files and reformat them.   Thus I have been forced back to stone knives and bear skins (well using vim and doing everything from command line).   It has been a serious reduction in what I can do as refactoring is out of the question with such a setup.  
> 
> It appears that netbeans has finally readded C++ support, so I will try to get it spun up again.   

I was initially doing it in a similar fashion with having to rebuild the extension using `pip install -e .` and having to recompile everything every single time. I eventually just dropped in a cmake file so I can build the native code with ninja and a gradle script for the java. It's much better not having to wait two minutes to rebuild everything for a minor change. IMO, this is mostly just a side effect of how python builds c extensions in a platform independent way.

Reply:
It wouldn't be much of an effort for them to make an option like my "python setup.py build_ext --makefile" which just takes all the instructions they are running on a system and dumping it out in makefile format.   I would never have made it this far in the project without it.   Unfortunately with the motion towards a more opaque process it is getting harder to accomplish such development.

Reply:
> It wouldn't be much of an effort for them to make an option like my "python setup.py build_ext --makefile" which just takes all the instructions they are running on a system and dumping it out in makefile format.   I would never have made it this far in the project without it.   Unfortunately with the motion towards a more opaque process it is getting harder to accomplish such development.

fwiw cmake does a pretty decent job with python extensions. The only thing that was lacking when I set it up was free threaded python support. It might be able to generate the netbeans project for you too.

Reply:
I'd be in favor of tools which are independent of the used IDE. Otherwise everybody would be forced to use the same IDE, right?

Reply:
> I'd be in favor of tools which are independent of the used IDE. Otherwise everybody would be forced to use the same IDE, right?

I'm always fighting someone to make something os/ide independent, even at work. Everyone should be able to use what they're most comfortable and productive with.

Reply:
I just want a beautifier to run from command line when we check in.   So long as the style is close enough that anyone's personal IDE doesn't go insane trying process it then all is good.  Double points if the beautifier is a python module so that we can just pull it in with requirements rather than installing another tool.  I distributed the Netbeans IDE files in the project directory, but there is no expectation that anyone needs to use them.

Reply:
Right, so lets run pre-commit on PRs, so everybody can do whatever they love offline. When the code goes online, we enforce the formatting (this can be done automatically, but would involve rebasing to origin after a CI run).

It is also called evolution when the young ones wants to introduce state of the art stuff. I was also young one day and my bosses hated me for introducing SVN over CVS =) Sooo complicated. Several years later we had the same thingy with Git. Now everybody uses it. So after all I wouldn't say it's a bad thing to modernize once in a while.

The productivity drop is preliminary and would probably benefit in the future.